fuel-vm/src/pool.rs

```diff
@@ -1,17 +1,17 @@
 //! Pool of VM memory instances for reuse.
 
-use crate::interpreter::{
-    Memory,
-    MemoryInstance,
-};
+use crate::interpreter::Memory;
+
+#[cfg(any(test, feature = "test-helpers"))]
+use crate::interpreter::MemoryInstance;
 
 /// Trait for a VM memory pool.
 pub trait VmMemoryPool: Sync {
     /// The memory instance returned by this pool.
     type Memory: Memory + Send + Sync + 'static;
 
     /// Gets a new VM memory instance from the pool.
-    fn get_new(&self) -> Self::Memory;
+    fn get_new(&self) -> impl core::future::Future<Output = Self::Memory> + Send;
```

Comment:
It is not related to the change, but on `fuel-core` side we are using async function, not sync.